projects/fourier/lib/fourier/utilities/system.rb

```diff
@@ -9,17 +9,17 @@ def self.system(*args)
         Kernel.system(*args) || exit(1)
       end
 
-      def self.tuist(*args, **kwargs)
-        @tuist = "/usr/local/bin/tuist"
+      def self.tuist(*args, from_source: false)
+        tuist = "/usr/local/bin/tuist"
 
-        if kwargs[:from_source] || !File.exist?(@tuist)
```

Comment:
If I am not mistaken, running `tuist` from source via `fourier` was possible before, too? Could you maybe describe a bit the meaning behind this PR? I might be missing something here 🙏

Reply:
#3950 Made it use the system installed tuist to make it easier for people to contribute without having to build tuist from source to get the project running.

The `fourier tuist` command still defaults to building from source. @danyf90 mentioned adding a flag to allow the other command to build from source if you wanted to test features that way.

Thinking about it more im not sure which default is best?

Reply:
ah, I see, thanks for the context!

I think the current default of using global tuist is better since you can still generate a project even if your local version of tuist is uncompilable.

Reply:
But you can already do that by invoking global tuist directly.

On the other hand, with fourier using global tuist it can fail even if your changes are valid. For example, on the 3.0 branch fourier will fail when trying to using global tuist 2.x.

Reply:
That's why we are adding the --from-source flag. To be able to use the command when the global one is failing

Reply:
I was arguing that from source is a better default.

But perhaps we can do both as the default behavior. Try it one way, and if it returns an error try it the other.

Reply:
99% of the times we won't have breaking changes (it happens only right before a major release), so I think using the system one is a better default as it saves a lot of time by not needing to build tuist